packages/go_router/lib/src/misc/extensions.dart

```diff
@@ -79,8 +91,16 @@ extension GoRouterHelper on BuildContext {
   /// * [replace] which replaces the top-most page of the page stack but treats
   ///   it as the same page. The page key will be reused. This will preserve the
   ///   state and not run any page animation.
-  void pushReplacement(String location, {Object? extra}) =>
-      GoRouter.of(this).pushReplacement(location, extra: extra);
+  void pushReplacement(
```

Comment:
just took a quick glance, should we make this return future instead of passing in a completer?

Reply:
@chunhtai 

If we have page A -> push -> page B -> pushReplacement -> page C,

this still won't solve the problem because although we can track when the completer from pushReplacement finishes on page B by returning a Future, the Future on page A will never complete.

In the _updateRouteMatchList function, in the case of NavigatingType.pushReplacement, the completer responsible for the future in the push method is simply removed and never completed.

Passing a completer through the push method allows us to detect when such a page has been replaced.

Reply:
I think the this is intended? the first push will never complete since it was replaced by another page regardless whether you pass in the completer or not